internal/provider/apikey_resource_test.go

```diff
@@ -0,0 +1,44 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"testing"
+)
+
+func createRandomApiKeyName() string {
+	return fmt.Sprintf("key-terraformprovider-%s", randomString())
+}
+
+func TestAccBasicApiKey(t *testing.T) {
+	apiKeyName := createRandomApiKeyName()
+	config := func(displayName string, ownerType string, ownerId string) string {
+		return fmt.Sprintf(`
+provider "temporalcloud" {
+
+}
+
+resource "temporalcloud_apikey" "terraform" {
+  display_name = "%s"
+	  owner_type = "%s"
+	  owner_id = "%s"
+	expiry_time = "2024-10-01T00:00:00Z"
+	description = "Test API Key"
```

Comment:
If I don't supply description I get error
```
When applying changes to temporalcloud_apikey.terraform, provider
        "provider[\"registry.terraform.io/hashicorp/temporalcloud\"]" produced an
        unexpected new value: .description: was null, but now cty.StringVal("").
```

Reply:
hmm - that should be an option field.  TY for the feedback. We will look into this.

Reply:
I made it optional only but I believe its because of protobuf (default string value is empty). Let me know if this is something I need to handle in the provider

Reply:
I handled it in the create method